yarl/_url.py

```diff
@@ -32,8 +32,8 @@
 
 import idna
 from multidict import MultiDict, MultiDictProxy, istr
+from propcache import under_cached_property as cached_property
```

Comment:
Why is it `under_`, though?

Reply:
because it uses `self._cache` to store the cache

Reply:
oh